crates/bevy_scene/src/scene.rs

```diff
@@ -18,3 +25,68 @@ impl Scene {
         Self { world }
     }
 }
+
+impl Scene {
+    /// Clone the scene.
+    ///
+    /// This method will return a `SceneSpawnError` if either a type is not registered in the
+    /// provided `AppTypeRegistry` or doesn't reflect the `Component` trait.
+    pub fn clone(&self, type_registry: &AppTypeRegistry) -> Result<Scene, SceneSpawnError> {
+        let mut new_world = World::new();
+        new_world.insert_resource(type_registry.clone());
+        self.write_to_world(&mut new_world)?;
+        new_world.remove_resource::<AppTypeRegistry>();
+        Ok(Self { world: new_world })
+    }
+
+    /// Write the entities and their corresponding components to the given world.
+    ///
+    /// This method will return a `SceneSpawnError` if either a type is not registered
+    /// or doesn't reflect the `Component` trait.
+    pub fn write_to_world(&self, world: &mut World) -> Result<InstanceInfo, SceneSpawnError> {
+        let mut instance_info = InstanceInfo {
+            entity_map: EntityMap::default(),
+        };
+
+        let type_registry = world.resource::<AppTypeRegistry>().clone();
```

Comment:
Can't we just pass the type registry as argument? Especially since this forces the little dance of add/remove in `clone()`?

Reply:
I wanted an api similar to https://docs.rs/bevy/latest/bevy/scene/struct.DynamicScene.html#method.write_to_world but I guess no one use that one anyway